### ISSUE TITLE
Revamp helloWorld and stress tests

### ISSUE THREAD

**Issue by hannes-ucsc** [BD2KGenomics/toil#210]
Running helloWorld.py on Mesos in Ec2 produces this output

```
ubuntu@ip-172-31-16-138:~$ python helloWorld.py --toil=aws:us-west-2:hannes-foo --batchSystem=mesos --masterIP=mesos-master:5050
INFO:toil.lib.bioio:Logging set at level: INFO
INFO:toil.common:Starting to create the toil setup for the first time
INFO:toil.common:Using the mesos batch system
ERROR:boto:400 Bad Request
ERROR:boto:<?xml version="1.0"?>
<Response><Errors><Error><Code>NoSuchDomain</Code><Message>The specified domain does not exist.</Message><BoxUsage>0.0000137200</BoxUsage></Error></Errors><RequestID>2d473c16-5eb7-792f-e23b-7bf2626b1b28</RequestID></Response>
ERROR:boto:400 Bad Request
ERROR:boto:<?xml version="1.0"?>
<Response><Errors><Error><Code>NoSuchDomain</Code><Message>The specified domain does not exist.</Message><BoxUsage>0.0000137200</BoxUsage></Error></Errors><RequestID>0531988f-a5f4-0e7f-13bf-58b954901164</RequestID></Response>
ERROR:boto:400 Bad Request
ERROR:boto:<?xml version="1.0"?>
<Response><Errors><Error><Code>NoSuchDomain</Code><Message>The specified domain does not exist.</Message><BoxUsage>0.0000137200</BoxUsage></Error></Errors><RequestID>9463062c-0c5a-21a2-6ff5-0a61a022caf8</RequestID></Response>
WARNING:toil.jobStores.awsJobStore:Creation of 'hannes-foo--versions' still pending, retrying in 5s
I0805 05:49:25.919760  2433 sched.cpp:157] Version: 0.22.0
I0805 05:49:25.921713  2446 sched.cpp:254] New master detected at master@172.31.16.138:5050
I0805 05:49:25.921983  2446 sched.cpp:264] No credentials provided. Attempting to register without authentication
I0805 05:49:25.923115  2446 sched.cpp:448] Framework registered with 20150804-214747-2316312492-5050-1535-0007
WARNING:toil.batchSystems.mesos.batchSystem:No jobs to assign. Rejecting offer
INFO:toil.common:Written the environment for the jobs to the environment file
INFO:toil.leader:Checked batch system has no running jobs and no updated jobs
INFO:toil.leader:Found 1 jobs to start and 0 jobs with successors to run
INFO:toil.leader:Starting the main loop
INFO:toil.batchSystems.mesos.batchSystem:Preparing to launch Mesos task 0 using offer 20150804-214747-2316312492-5050-1535-O20...
INFO:toil.batchSystems.mesos.batchSystem:...launching Mesos task 0
INFO:toil.batchSystems.mesos.batchSystem:Executor value: "toilExecutor"
 on slave value: "20150804-214747-2316312492-5050-1535-S0"
 sent a message: cpu percent: 100.0, ram usage: 14.1
INFO:toil.leader:Only failed jobs and their dependents (0 total) are remaining, so exiting.
INFO:toil.leader:Finished the main loop
INFO:toil.leader:Waiting for stats and logging collator process to finish
WARNING:toil.batchSystems.mesos.batchSystem:No jobs to assign. Rejecting offer
INFO:toil.leader:Stats/logging finished collating in 1.66737389565 seconds
INFO:toil.batchSystems.mesos.batchSystem:Stopping Mesos driver
I0805 05:49:36.894577  2433 sched.cpp:1589] Asked to stop the driver
INFO:toil.batchSystems.mesos.batchSystem:Joining Mesos driver
INFO:toil.batchSystems.mesos.batchSystem:Joined Mesos driver
I0805 05:49:36.894726  2441 sched.cpp:831] Stopping framework '20150804-214747-2316312492-5050-1535-0007'
```

No error message, etc. Nothing in executor's stdout/stderr. However, the job store files that are created by the test don't exist. I'm not sure if this is intended.

**Comment by hannes-ucsc** [comment]
Related to #234

**Comment by cket** [comment]
Closed by 7890363a6c37c225b652c303b746057b6ac980f2

**Comment by paul-sanghera** [comment]
Running toil-sort-example.py (from toil doc)  on Mesos in Ec2 produces this output:
mesosbox@ip-172-31-16-87:~$  python2.7 toil-sort-example.py  --batchSystem=mesos --mesosMaster=mesos-master:5050  aws:us-west-2:my-jobstore
ip-172-31-16-87: 2017-06-15 17:54:06,215 INFO: toil.lib.bioio: Logging set at level: INFO
ip-172-31-16-87: 2017-06-15 17:54:06,215 INFO: toil.lib.bioio: Logging set at level: INFO
ip-172-31-16-87: 2017-06-15 17:54:06,303 INFO: toil.jobStores.aws.jobStore: Binding to job store domain 'toil-registry'.
ip-172-31-16-87: 2017-06-15 17:54:06,366 INFO: toil.jobStores.aws.jobStore: Binding to job store domain 'toil-registry'.
ip-172-31-16-87: 2017-06-15 17:54:06,406 INFO: toil.jobStores.aws.jobStore: Binding to job store domain 'my-jobstore--jobs'.
ip-172-31-16-87: 2017-06-15 17:54:06,419 ERROR: boto: 400 Bad Request
ip-172-31-16-87: 2017-06-15 17:54:06,420 ERROR: boto: <?xml version="1.0"?>
<Response><Errors><Error><Code>NoSuchDomain</Code><Message>The specified domain does not exist.</Message><BoxUsage>0.0000137200</BoxUsage></Error></Errors><RequestID>b9277ec0-e107-55cd-dab3-8cdeb742a1da</RequestID></Response>
ip-172-31-16-87: 2017-06-15 17:54:07,505 INFO: toil.jobStores.aws.jobStore: Binding to job store domain 'my-jobstore--files'.
ip-172-31-16-87: 2017-06-15 17:54:07,553 ERROR: boto: 400 Bad Request
ip-172-31-16-87: 2017-06-15 17:54:07,554 ERROR: boto: <?xml version="1.0"?>
<Response><Errors><Error><Code>NoSuchDomain</Code><Message>The specified domain does not exist.</Message><BoxUsage>0.0000137200</BoxUsage></Error></Errors><RequestID>880e2916-e60a-b154-f90c-014423995c80</RequestID></Response>
ip-172-31-16-87: 2017-06-15 17:54:08,641 INFO: toil.jobStores.aws.jobStore: Binding to job store bucket 'my-jobstore--files'.
ip-172-31-16-87: 2017-06-15 17:54:11,496 INFO: toil.jobStores.abstractJobStore: The workflow ID is: '3604e1cb-a103-4278-b645-ebefad8d1624'
ip-172-31-16-87: 2017-06-15 17:54:11,823 INFO: toil.jobStores.aws.jobStore: Binding to job store domain 'toil-registry'.
ip-172-31-16-87: 2017-06-15 17:54:12,054 INFO: toil.common: Using the mesos batch system
I0615 17:54:12.329828  2438 sched.cpp:164] Version: 0.25.0
I0615 17:54:12.332041  2450 sched.cpp:262] New master detected at master@172.31.16.87:5050
I0615 17:54:12.332224  2450 sched.cpp:272] No credentials provided. Attempting to register without authentication
I0615 17:54:12.333691  2447 sched.cpp:641] Framework registered with e2c3a9b8-026f-4b6c-9c6b-1b0e60519d93-0003
ip-172-31-16-87: 2017-06-15 17:54:12,371 INFO: toil.common: Written the environment for the jobs to the environment file
ip-172-31-16-87: 2017-06-15 17:54:12,372 INFO: toil.common: Caching all jobs in job store
ip-172-31-16-87: 2017-06-15 17:54:12,657 INFO: toil.common: 0 jobs downloaded.
ip-172-31-16-87: 2017-06-15 17:54:12,895 INFO: toil.realtimeLogger: Real-time logging disabled
ip-172-31-16-87: 2017-06-15 17:54:12,897 INFO: toil.leader: (Re)building internal scheduler state
ip-172-31-16-87: 2017-06-15 17:54:12,897 INFO: toil.leader: Checked batch system has no running jobs and no updated jobs
ip-172-31-16-87: 2017-06-15 17:54:12,897 INFO: toil.leader: Found 1 jobs to start and 0 jobs with successors to run
ip-172-31-16-87: 2017-06-15 17:54:12,900 INFO: toil.leader: Starting the main loop
ip-172-31-16-87: 2017-06-15 17:54:18,001 INFO: toil.batchSystems.mesos.batchSystem: Preparing to launch Mesos task 0 using offer e2c3a9b8-026f-4b6c-9c6b-1b0e60519d93-O19 ...
ip-172-31-16-87: 2017-06-15 17:54:18,001 INFO: toil.batchSystems.mesos.batchSystem: Launched Mesos task 0.
ip-172-31-16-87: 2017-06-15 17:54:18,002 INFO: toil.batchSystems.mesos.batchSystem: Although there are queued jobs, none of them could be run with theoffers extended to the framework. Enable debug logging to see details about tasks queued and offers made.
ip-172-31-16-87: 2017-06-15 17:54:21,816 WARNING: toil.leader: The jobWrapper seems to have left a log file, indicating failure: 277e5001-4df8-4660-a088-9c6234b0f97f
ip-172-31-16-87: 2017-06-15 17:54:21,832 WARNING: toil.leader: Reporting file: 277e5001-4df8-4660-a088-9c6234b0f97f
ip-172-31-16-87: 2017-06-15 17:54:21,832 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	---TOIL WORKER OUTPUT LOG---
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	WARNING:toil.resource:The localize() method should only be invoked on a worker.
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	WARNING:toil.resource:The localize() method should only be invoked on a worker.
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	Traceback (most recent call last):
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/worker.py", line 330, in main
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    job._runner(jobWrapper=jobWrapper, jobStore=jobStore, fileStore=fileStore)
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1043, in _runner
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    returnValues = self._run(jobWrapper, fileStore)
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 991, in _run
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    return self.run(fileStore)
ip-172-31-16-87: 2017-06-15 17:54:21,833 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 1136, in run
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    rValue = userFunction(*((self,) + tuple(self._args)), **self._kwargs)
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/home/mesosbox/toil-sort-example.py", line 15, in setup
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/fileStore.py", line 241, in writeGlobalFile
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    jobStoreFileID = self.jobStore.writeFile(absLocalFileName, cleanupID)
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 498, in writeFile
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    info.upload(localFilePath)
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 970, in upload
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    file_size, file_time = self._fileSizeAndTime(localFilePath)
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	  File "/usr/local/lib/python2.7/dist-packages/toil/jobStores/aws/jobStore.py", line 1214, in _fileSizeAndTime
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	    file_stat = os.stat(localFilePath)
ip-172-31-16-87: 2017-06-15 17:54:21,834 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	OSError: [Errno 2] No such file or directory: '/home/mesosbox/file_to_sort.txt'
ip-172-31-16-87: 2017-06-15 17:54:21,835 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	Exiting the worker because of a failed jobWrapper on host ip-172-31-27-21
ip-172-31-16-87: 2017-06-15 17:54:21,835 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	ERROR:toil.worker:Exiting the worker because of a failed jobWrapper on host ip-172-31-27-21
ip-172-31-16-87: 2017-06-15 17:54:21,835 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	WARNING:toil.jobWrapper:Due to failure we are reducing the remaining retry count of job 277e5001-4df8-4660-a088-9c6234b0f97f to 0
ip-172-31-16-87: 2017-06-15 17:54:21,835 WARNING: toil.leader: 277e5001-4df8-4660-a088-9c6234b0f97f:	WARNING:toil.jobWrapper:We have increased the default memory of the failed job to 2147483648 bytes
ip-172-31-16-87: 2017-06-15 17:54:21,835 WARNING: toil.leader: Job: 277e5001-4df8-4660-a088-9c6234b0f97f is completely failed
ip-172-31-16-87: 2017-06-15 17:54:21,835 INFO: toil.leader: No jobs left to run so exiting.
ip-172-31-16-87: 2017-06-15 17:54:21,835 INFO: toil.leader: Finished the main loop
ip-172-31-16-87: 2017-06-15 17:54:21,835 INFO: toil.leader: Waiting for stats and logging collator process to finish ...
ip-172-31-16-87: 2017-06-15 17:54:22,321 INFO: toil.leader: ... finished collating stats and logs. Took 0.48540687561 seconds
ip-172-31-16-87: 2017-06-15 17:54:22,321 INFO: toil.leader: Waiting for service manager thread to finish ...
ip-172-31-16-87: 2017-06-15 17:54:22,899 INFO: toil.leader: ... finished shutting down the service manager. Took 0.577798128128 seconds
ip-172-31-16-87: 2017-06-15 17:54:22,911 INFO: toil.leader: Finished toil run with 1 failed jobs
ip-172-31-16-87: 2017-06-15 17:54:22,911 INFO: toil.leader: Failed jobs at end of the run: set(['277e5001-4df8-4660-a088-9c6234b0f97f'])
ip-172-31-16-87: 2017-06-15 17:54:22,911 INFO: toil.batchSystems.mesos.batchSystem: Stopping Mesos driver
I0615 17:54:22.911774  2438 sched.cpp:1771] Asked to stop the driver
I0615 17:54:22.911885  2452 sched.cpp:1040] Stopping framework 'e2c3a9b8-026f-4b6c-9c6b-1b0e60519d93-0003'
ip-172-31-16-87: 2017-06-15 17:54:22,911 INFO: toil.batchSystems.mesos.batchSystem: Joining Mesos driver
ip-172-31-16-87: 2017-06-15 17:54:22,912 INFO: toil.batchSystems.mesos.batchSystem: Joined Mesos driver
Traceback (most recent call last):
  File "toil-sort-example.py", line 172, in <module>
    main()
  File "toil-sort-example.py", line 169, in main
    memory='1000M'), options)
  File "/usr/local/lib/python2.7/dist-packages/toil/job.py", line 500, in startToil
    return toil.start(job)
  File "/usr/local/lib/python2.7/dist-packages/toil/common.py", line 562, in start
    return self._runMainLoop(job)
  File "/usr/local/lib/python2.7/dist-packages/toil/common.py", line 791, in _runMainLoop
    jobCache=self._jobCache)
  File "/usr/local/lib/python2.7/dist-packages/toil/leader.py", line 712, in mainLoop
    raise FailedJobsException( config.jobStore, len(toilState.totalFailedJobs) )
toil.leader.FailedJobsException: The job store 'aws:us-west-2:my-jobstore' contains 1 failed jobs
